### PR TITLE
task(0101): Remove the legacy "risk classification" fields (`IsNetwork` and `Category`) from syscall and symbol analysis

### PR DESCRIPTION
**Pull request by isseis**
# Pull Request: Removal of Record-Level Risk Classification

This pull request finalizes the removal of record-level risk classification fields (`IsNetwork` and `Category`) from syscall and symbol analysis results, and updates all related logic, documentation, and tests to use runtime-derived network classification instead.

The changes ensure that network-related detection is now based on runtime checks (such as syscall number and symbol name) rather than persisted fields, and that logs maintain equivalent information by deriving categories on the fly. All implementation plan checklists are marked as complete, and the codebase is updated to pass lint, build, and test checks.

---

### 1. Core logic and schema changes
* Removed the `IsNetwork` field from `SyscallInfo` and the `Category` field from `DetectedSymbolEntry`, updating all code, documentation, and tests to reflect this change.
* Network classification is now performed at runtime using syscall numbers and symbol names.
* Updated the schema version to `18` and adjusted schema migration and version checks accordingly.

**References:**
* [[1]](diffhunk://#diff-8b2849d7be20065f3bbba05f2aaf08577c01f802b70a03f22f07cb0e69838114L73-R77)
* [[2]](diffhunk://#diff-5da921ce724ec48896630d7fc28e08d019ca89a0861c876ea2a7e2a28e98239bL529-R554)

### 2. Network classification improvements
* Refactored network detection logic to use syscall number tables and symbol name helpers at runtime (`IsNetworkSyscall`, `IsNetworkSymbolName`).
* Added fallback category assignment for symbols:
  * `dynamic_load`: for dynamic load symbols.
  * `syscall_wrapper`: for others.
* Updated architecture diagrams, specifications, and documentation to reflect the new runtime-based classification logic.

**References:**
* [[1]](diffhunk://#diff-8b2849d7be20065f3bbba05f2aaf08577c01f802b70a03f22f07cb0e69838114L117-R123)
* [[2]](diffhunk://#diff-bcade0f37a1b2804f04c99f8599c13d00008a601101be3bacf387fd90a7ee55cL122-R125)

### 3. Testing and validation
* Added and updated tests to verify:
  * Absence of removed fields in JSON outputs.
  * Correct behavior of runtime classification.
  * Schema versioning and handling of edge cases (unknown architectures, negative syscall numbers, nil results).
* Marked all implementation plan checkboxes as complete.

**References:**
* [[1]](diffhunk://#diff-5da921ce724ec48896630d7fc28e08d019ca89a0861c876ea2a7e2a28e98239bL468-R479)
* [[2]](diffhunk://#diff-8b2849d7be20065f3bbba05f2aaf08577c01f802b70a03f22f07cb0e69838114L185-R190)

### 4. Codebase and tooling updates
* Updated `.golangci.yml` to include the new `internal/libccache` directory in linter checks.

**References:**
* [[1]](diffhunk://#diff-6179837f7df53a6f05c522b6b7bb566d484d5465d9894fb04910dd08bb40dcc9R52)

---

These changes modernize the risk classification approach, improve maintainability, and ensure consistent, runtime-driven detection across the codebase.

### CODE REVIEW

File: internal/runner/security/network_analyzer.go [hunk 1]
```diff
@@ -19,6 +20,25 @@ import (
...
+
+	switch arch {
+	case "x86_64":
+		return elfanalyzer.NewX86_64SyscallTable()
+	case "arm64":
+		return elfanalyzer.NewARM64LinuxSyscallTable()
+	default:
+		return nil
+	}
+}
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/network_analyzer.go:40]: ご指摘の通り、 は実行ホストの OS に依存しており、Linux で記録された ELF バイナリを macOS 上で検査する場合に誤ったテーブルが使用される可能性があります。この問題を根本的に解決するには  スキーマに OS フィールドを追加する必要があり、本 PR のスコープを超えます。別 issue として追跡します。

[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/network_analyzer.go:40]: 対応しました。x86_64・arm64・macOS 用の syscall テーブルをパッケージレベルの変数 (cachedX86Table, cachedArm64Table, cachedMacOSTable) としてキャッシュするよう修正しました。

File: internal/runner/security/elfanalyzer/standard_analyzer.go [hunk 2]
```diff
@@ -426,3 +427,14 @@ func (a *StandardELFAnalyzer) convertSyscallResult(result *SyscallAnalysisResult
...
+func syscallTableForArchitecture(arch string) SyscallNumberTable {
+	switch arch {
+	case "x86_64":
+		return NewX86_64SyscallTable()
+	case "arm64":
+		return NewARM64LinuxSyscallTable()
+	default:
+		return nil
+	}
+}
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/elfanalyzer/standard_analyzer.go:440]: 対応しました。elfanalyzer パッケージ内でも x86_64・arm64 用テーブルをパッケージレベルの変数 (cachedX86SyscallTable, cachedArm64SyscallTable) としてキャッシュするよう修正しました。

File: internal/runner/security/network_analyzer.go [hunk 3]
```diff
@@ -297,15 +317,19 @@ func syscallAnalysisHasSVCSignal(result *fileanalysis.SyscallAnalysisResult) boo
...
 // This includes resolved svc entries (DeterminationMethod == "direct_svc_0x80" AND Number != -1)
 // whose network classification is determined by the syscall table lookup.
 func syscallAnalysisHasNetworkSignal(result *fileanalysis.SyscallAnalysisResult) bool {
 	if result == nil {
 		return false
 	}
+	table := syscallTableForArch(result.Architecture)
+	if table == nil {
+		return false
+	}
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/network_analyzer.go:330]: 対応しました。DetectedSyscalls が空の場合に早期リターンするチェックを syscallTableForArch の呼び出しより前に追加し、不要なテーブルルックアップを省略するよう修正しました。
